src/StaticWebAssetsSdk/Targets/Microsoft.NET.Sdk.StaticWebAssets.JSModules.targets

```diff
@@ -40,6 +40,17 @@ Copyright (c) .NET Foundation. All rights reserved.
       ResolveJsInitializerModuleStaticWebAssets;
     </ResolveCoreStaticWebAssetsDependsOn>
 
+    <ResolveJSModuleStaticWebAssetsDependsOn>
+      $(ResolveJSModuleStaticWebAssetsDependsOn);
+      ResolveProjectStaticWebAssets;
+    </ResolveJSModuleStaticWebAssetsDependsOn>
+
+    <ResolveJSModuleStaticWebAssetsDependsOn Condition="'$(UsingMicrosoftNETSdkRazor)' == 'true'">
+      $(ResolveJSModuleStaticWebAssetsDependsOn);
+      ResolveRazorComponentInputs;
+      ResolveRazorGenerateInputs;
+    </ResolveJSModuleStaticWebAssetsDependsOn>
```

Comment:
Is this the same pattern that we follow on Scoped CSS?

Reply:
VS runs the `UpdateStaticWebAssetsDesignTime` target during hot reload, which depends on two other targets:
* `ResolveReferencedProjectsStaticWebAssets`
* `GenerateComputedBuildStaticWebAssets`

Scoped CSS targets run as dependencies of the `GenerateComputedBuildStaticWebAssets` target. However, the `ResolveJSModuleStaticWebAssets` target was running as part of `ResolveReferencedProjectsStaticWebAssets`. So, adding `ResolveRazorComponentInputs` and `ResolveRazorGenerateInputs` to `GenerateComputedBuildStaticWebAssetsDependsOn` (as we do with scoped CSS) doesn't fix the issue because it doesn't guarantee that the targets run in the right order.

We could probably still follow the same pattern as scoped CSS, but my guess is we'd probably still need to add the two targets as an `AfterTargets` on `ResolveJSModuleStaticWebAssets`.

Reply:
I don't think I follow, let's have a quick chat and see how we can make sure that both scoped CSS and JS modules follow the exact same pattern.

If in the future we add another thing like this, we want to have a clear pattern that showcases how to integrate it.